testsuite/tests/kuadrant/authorino/conditions/test_patternref_expressions.py

```diff
@@ -0,0 +1,80 @@
+"""Test patterns reference functionality and All/Any logical expressions."""
+import pytest
+
+from testsuite.objects import Pattern, PatternRef, AnyPatterns, AllPatterns
+
+
+@pytest.fixture(scope="module")
+def authorization(authorization):
+    """
+    Add multiple named patterns to AuthConfig to be referenced in later authorization rules.
+    Create authorization rule which:
+    1. For a GET requests allows only paths "/anything/dog" and "/anything/cat"
+    2. For a POST requests allows only paths "/anything/apple" and "/anything/pear"
+    3. For requests that contain header "x-special" than it is authorized regardless.
+    """
+    authorization.add_patterns(
+        {
+            "apple": [Pattern("context.request.http.path", "eq", "/anything/apple")],
+            "pear": [Pattern("context.request.http.path", "eq", "/anything/pear")],
+            "dog": [Pattern("context.request.http.path", "eq", "/anything/dog")],
+            "cat": [Pattern("context.request.http.path", "eq", "/anything/cat")],
+            "get": [Pattern("context.request.http.method", "eq", "GET")],
+            "post": [Pattern("context.request.http.method", "eq", "POST")],
+        }
+    )
+
+    authorization.authorization.add_auth_rules(
+        "auth_rules",
+        [
+            AnyPatterns(
+                [
+                    AllPatterns([AnyPatterns([PatternRef("dog"), PatternRef("cat")]), PatternRef("get")]),
+                    AllPatterns([AnyPatterns([PatternRef("apple"), PatternRef("pear")]), PatternRef("post")]),
+                    Pattern("context.request.http.headers.@keys", "incl", "x-special"),
+                ]
+            )
+        ],
+    )
+
+    return authorization
+
+
+def get_evaluator(suffix: str):
+    """Expected return value for suffix in /anything/{suffix} with GET request."""
+    if suffix in {"dog", "cat"}:
+        return 200
+    return 403
+
+
+def post_evaluator(suffix: str):
```

Comment:
how about parametrizing directly the test? that way you do not need this function

Reply:
Changed the parametrization on directly on test.